src/game/ai/alligator.c

```diff
@@ -105,11 +106,19 @@ void AlligatorControl(int16_t item_num)
     case ALLIGATOR_SWIM:
         if (info.bite && item->touch_bits) {
             item->goal_anim_state = ALLIGATOR_ATTACK;
+            if (g_Config.fix_alligator_ai) {
+                item->required_anim_state = ALLIGATOR_SWIM;
+            }
         }
         break;
 
     case ALLIGATOR_ATTACK:
-        if (item->frame_number == g_Anims[item->anim_number].frame_base) {
+        if (g_Config.fix_alligator_ai
+            && item->frame_number == ALLIGATOR_BITE_AF) {
+            item->required_anim_state = ALLIGATOR_EMPTY;
+        } else if (
+            !g_Config.fix_alligator_ai
+            && item->frame_number == g_Anims[item->anim_number].frame_base) {
```

Comment:
I believe these can be simplified to
`if (frame == frame_base (fix croc && frame == alligator_bite_af))`

Reply:
I don't think so because the fix version will have the extra damage bug because it can also pass the `item->frame_number == g_Anims[item->anim_number].frame_base` check which is true for multiple animations.

Reply:
Right, then how about `if (frame == fix alligator ? ALLIGATOR_BITE_AF : frame_base) {`?